.github/workflows/maven.yml

```diff
@@ -0,0 +1,35 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
```

Comment:
I notice that CI jobs did not trigger for this PR. I do not think it's becase of a problem with the workflow, but rather because `GitHub Actions` are not enabled on this repo. This would normally be done through a ticket but alternatively we can probably do it using the new self-service thing put in place by the Foundation. 

Since this is just the initial contribution, I think this could be done after the PR is merged - the build and tests work locally.